src/libpgmoneta/wal.c

```diff
@@ -379,6 +383,11 @@ pgmoneta_wal(int srv, char** argv)
                         break;
                      }
                   }
+                  // assign xlogpos to LSN after a message data is written to the segment
+                  xlogpos = wal_xlogptr_to_xlogpos(xlogptr);
```

Comment:
We don't need `xlogpos` anymore since we have `config->servers[srv].current_wal_lsn`

Reply:
Removed it.